apps/oc_chef_wm/src/oc_chef_wm_named_policy.erl

```diff
@@ -0,0 +1,205 @@
+    {['GET', 'PUT', 'POST', 'DELETE'], Req, State}.
+
+post_is_create(Req, State) ->
+    {true, Req, State}.
+
+create_path(Req, State) ->
+    Name = wrq:path_info(policy_name, Req),
+    {Name, Req, State}.
+
+validate_request(Method, Req, State = #base_state{organization_guid = OrgId}) when Method == 'GET';
+                                          Method == 'DELETE' ->
+    Name = wrq:path_info(policy_name, Req),
+    Group = wrq:path_info(policy_group, Req),
+    {Req, State#base_state{superuser_bypasses_checks = true, resource_state = #policy_state{oc_chef_policy = #oc_chef_policy{org_id = OrgId, name = Name, policy_group = Group}}}};
+validate_request(_PostOrPut, Req, #base_state{organization_guid = OrgId, resource_state = PolicyState} = State) ->
+    Policy = validate_json(Req),
+    Group = wrq:path_info(policy_group, Req),
+    NewPolicy = ej:set({<<"policy_group">>}, Policy, erlang:list_to_binary(Group)),
+    {Req, State#base_state{
+            superuser_bypasses_checks = true,
+            resource_state = PolicyState#policy_state{
+                               oc_chef_policy = #oc_chef_policy{org_id = OrgId, policy_group = Group},
+                               policy_data = NewPolicy}}}.
+
+validate_json(Req) ->
+    Body = wrq:req_body(Req),
+    {ok, Policy} = oc_chef_policy:parse_binary_json(Body),
+    ok = validate_name(Req, Policy),
+    case ej:valid(?VALIDATION_CONSTRAINTS, Policy) of
+        ok ->
+            Policy;
+        Bad ->
+            throw(Bad)
+    end.
+
+validate_name(Req, Policy) ->
+    NameFromReq = wrq:path_info(policy_name, Req),
+    NameFromJson = erlang:binary_to_list(ej:get({<<"name">>}, Policy)),
+    case ibrowse_lib:url_encode(NameFromJson) =:= NameFromReq of
+        true ->
+            ok;
+        false ->
+            erlang:throw({error, {mismatch, {<<"Name">>, ignored, ignored}}})
+    end.
+
+auth_info(Req, #base_state{chef_db_context = DbContext,
+                           resource_state = PolicyState = #policy_state{policy_data = PolicyData},
+                           organization_guid = OrgId} =State) ->
+    PolicyName = chef_wm_util:extract_from_path(policy_name, Req),
+    case wrq:method(Req) of
+        PostOrPut when PostOrPut =:= 'POST'; PostOrPut =:= 'PUT' ->
+            case oc_chef_wm_policies:validate_policy_name(ej:get({<<"id">>}, PolicyData, ej:get({<<"name">>}, PolicyData))) of
+                valid ->
+                    validate_policy(true, OrgId, PolicyName, DbContext, Req, State, PolicyState);
+                missing ->
+                    validate_policy(true, OrgId, PolicyName, DbContext, Req, State, PolicyState);
+                invalid  ->
+                    oc_chef_wm_policies:policy_name_invalid(Req, State)
+            end;
+        _DeleteOrGet ->
+            validate_policy(false, OrgId, PolicyName, DbContext, Req, State, PolicyState)
+    end.
+validate_policy(CanCreate, OrgId, PolicyName, DbContext, Req,
```

Comment:
Speaking of names, maybe "WantsToCreate" instead of "CanCreate" would be a little more clear?

Tangential: create on PUT still just 'feels' odd to me...